server.go

```diff
@@ -217,6 +217,12 @@ func (s *Server) listenAndServe() error {
 	defer inputCh.Close()
 	defer outputCh.Close()
 
+	// Notify everyone that the server has started. Runs sequentially so there
```

Comment:
I think it's too early to put this here. If we cannot starat the consumers I would say that the server has actually started and we will return an error on line 234. I think the reason the code was located as one of the very last calls was to ensure that when the user function is called we "guarantee" that we've setup and established a working connection.